doc/CMakeLists.txt

```diff
@@ -6,25 +6,14 @@ project(Zephyr-Kernel-Doc LANGUAGES)
 set(NO_BOILERPLATE TRUE)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE} ..)
 
-# Find west to (optionally) process modules for Kconfig
-find_program(
-  WEST
-  west
-  )
-if(${WEST} STREQUAL WEST-NOTFOUND)
-  unset(WEST)
+find_program(WEST west)
+if(NOT WEST)
+  message(FATAL_ERROR "The 'west' command was not found")
```

Comment:
`west` is an optional tool for Zephyr development, and we should start requiring it for building docs when it's not required for building Zephyr itself.

Reply:
I disagree here, since we generate content based on west docstrings, so it has to be installed for that part to work correctly. What we can do is to remove the check from CMake, since we don't check for any other doc Python dependency in the end.

Reply:
> I disagree here, since we generate content based on west docstrings, so it has to be installed for that part to work correctly.

Could we avoid building the west api when west is not available ?
After all, we are informing users that west is optional, even in the west docs themselves:
https://docs.zephyrproject.org/latest/guides/west/without-west.html
https://docs.zephyrproject.org/latest/guides/modules.html#without-west

and nowhere is it stated that west is required for doc build.
Only flash and debug are specifically mentioned not to work when not having west.

Reply:
I imagine Sphinx will throw some warnings if doc build is run without west.

Reply:
I have opted to remove west check, which I think in the end makes more sense. Users will be able to run the docs build, however, the build will likely finish with some warnings.